File: executable/julia/style-bar.jl
using Plotly
Plotly.signin("test-runner", "9h29fe3l0x")

data = [
  [
    "x" => ["giraffes", "orangutans", "monkeys"], 
    "y" => [20, 14, 23], 
    "name" => "SF Zoo", 
    "marker" => [
      "line" => ["color" => "grey"], 
      "color" => "orange"
    ], 
    "type" => "bar"
  ], 
  [
    "x" => ["giraffes", "orangutans", "monkeys"], 
    "y" => [12, 18, 29], 
    "name" => "LA Zoo", 
    "marker" => [
      "line" => [
        "color" => "grey", 
        "width" => 3
      ], 
      "color" => "blue"
    ], 
    "type" => "bar"
  ]
]
layout = [
  "title" => "Animal Population", 
  "xaxis" => ["type" => "category"], 
  "yaxis" => ["title" => "# of animals (thousands)"], 
  "categories" => ["giraffes", "orangutans", "monkeys"], 
  "barmode" => "group", 
  "bargap" => 0.25, 
  "bargroupgap" => 0.3, 
  "orientation" => "v"
]

response = Plotly.plot([data], ["layout" => layout, "filename" => "style-bar", "fileopt" => "overwrite"])
plot_url = response["url"]

File: executable/r/style-bar.r
library(plotly)

p <- plotly(username='test-runner', key='9h29fe3l0x')

trace1 <- list(
  x = c("giraffes", "orangutans", "monkeys"), 
  y = c(20, 14, 23), 
  name = "SF Zoo", 
  marker = list(
    line = list(color = "grey"), 
    color = "orange"
  ), 
  type = "bar"
)
trace2 <- list(
  x = c("giraffes", "orangutans", "monkeys"), 
  y = c(12, 18, 29), 
  name = "LA Zoo", 
  marker = list(
    line = list(
      color = "grey", 
      width = 3
    ), 
    color = "blue"
  ), 
  type = "bar"
)
data <- list(trace1, trace2)
layout <- list(
  title = "Animal Population", 
  xaxis = list(type = "category"), 
  yaxis = list(title = "# of animals (thousands)"), 
  categories = c("giraffes", "orangutans", "monkeys"), 
  barmode = "group", 
  bargap = 0.25, 
  bargroupgap = 0.3, 
  orientation = "v"
)

response <- p$plotly(data, kwargs=list(layout=layout, filename="style-bar", fileopt="overwrite"))
url <- response$url
filename <- response$filename